### PR TITLE
Fix log level for "local-redirect service exists for frontend" error

### LINKED ISSUE #16400: Error should be info: `local-redirect service exists for frontend, skip update for svc lrp-demo-service`
@aditighag confirmed that the following error message should be info level: 
```
level=error msg="Error while inserting service in LB map" error="local-redirect service exists for frontend {{10.123.250.188 {UDP 69} 0} 0}, skip update for svc lrp-demo-service" k8sNamespace=default k8sSvcName=lrp-demo-service subsys=k8s-watcher
```

### PR DESCRIPTION
**Pull request by tbalthazar**
- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] A ll code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [ ] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [ ] Provide a title or release-note blurb suitable for the release notes.
- [ ] Thanks for contributing!

Fixes: #16400

```release-note
Fix log level for "local-redirect service exists for frontend" error
```

### CODE REVIEW

**Review by aditighag — CHANGES_REQUESTED** [review]
Thanks for the PR. You are missing the error check for CLI - http://github.com/cilium/cilium/blob/master/daemon/cmd/loadbalancer.go#L106. This is the only place where we should log it as an error, the rest should be converted to debug level.
I've added the appropriate release-note label.

**Comment by tbalthazar** [comment]
@aditighag Thanks for the quick feedback.
I [added some logging in the CLI](https://github.com/cilium/cilium/pull/21898/commits/7bb2079e404c25c64b06fec3a75662017922b7a2): I assumed we want to log every errors, not just `ErrLocalRedirectServiceExists`. Does that make sense? Or do you think we should only long that error?

I also [converted the log levels to debug](https://github.com/cilium/cilium/pull/21898/commits/cc81b0839281c296efcf3f82814a66a8e672d055).

**Review by aditighag — CHANGES_REQUESTED** [review]
Thanks! Please squash last two commits. Also, let's remove this log message exception from tests - https://github.com/cilium/cilium/blob/master/test/helpers/cons.go#L234. Please add this in a new commit. I'll trigger the e2e tests once you've pushed these changes.

File: daemon/cmd/loadbalancer.go [hunk 1]
```diff
@@ -109,6 +109,7 @@ func (h *putServiceID) Handle(params PutServiceIDParams) middleware.Responder {
 			fmt.Errorf("the service provided is already registered with ID %d, please use that ID instead of %d",
 				id, frontend.ID))
 	} else if err != nil {
+		log.WithError(err).Error("Error while inserting service in LB map")
```
Comment (aditighag) [inline daemon/cmd/loadbalancer.go]: Why is this needed? The caller should already log the error, no? 
You can easily test this using `cilium service update ...` command (first create a local-redirect service, and then try to create the same service with clusterIP type). Ping me on Slack if you need any help.

Comment (tbalthazar) [reply daemon/cmd/loadbalancer.go]: @aditighag I probably misinterpreted your review above:
> Thanks for the PR. You are missing the error check for CLI - [http://github.com/cilium/cilium/blob/master/daemon/cmd/loadbalancer.go#L106](https://github.com/cilium/cilium/blob/master/daemon/cmd/loadbalancer.go#L106). This is the only place where we should log it as an error, the rest should be converted to debug level.

I reproduced the error as you suggested:
```
# cilium service update --local-redirect --backends 1.2.3.4:8080 --frontend 5.6.7.8:8080 --id 42
Creating new service with id '42'
Added service with 1 backends
# cilium service update --backends 1.2.3.4:8080 --frontend 5.6.7.8:8080 --id 42
Updating existing service with id '42'
Error: Cannot add/update service: [PUT /service/{id}][500] putServiceIdFailure  local-redirect service exists for frontend {{5.6.7.8 {TCP 8080} 0} 42}, skip update for svc / &{BackendAddresses:[0xc000844fc0] Flags:0xc000847f80 FrontendAddress:0xc000844c40 ID:42 UpdateServices:false}
```

It's printing the error message to `stderr` and it exits with an error code:
https://github.com/cilium/cilium/blob/master/cilium/cmd/service_update.go#L223

Is there anything else we should do here?

**Comment by tbalthazar** [comment]
Thanks for your feedback @aditighag, I addressed your remarks, rebased my commits and removed the exception from the tests.
Please let me know if there's anything left to do, re: https://github.com/cilium/cilium/pull/21898#discussion_r1007997837

File: daemon/cmd/loadbalancer.go (hunk 1 above)
Comment (aditighag) [reply daemon/cmd/loadbalancer.go]: The error should already be logged so no need to do anything else. Sorry for the confusion.

**Review by aditighag — APPROVED** [review]
Thanks for the changes! LGTM. I've triggered the e2e tests. 

Edit: The sole failure is a flake, and unrelated to the PR changes. The changes are quite specific to LRP so marking it as ready for merge.

**Pull request merged**